.clang-tidy

```diff
@@ -9,6 +9,7 @@
 # performance-unnecessary-value-param: TODO(RSDK-3007) remove this and fix all lints.
 # performance-move-const-arg: TODO(RSDK-3019)
 # misc-unused-parameters: TODO(RSDK-3008) remove this and fix all lints.
+# misc-include-cleaner: TODO(RSDK-5479) this is overly finnicky, add IWYU support and fix.
```

Comment:
A _lot_ of errors were generated without this change. Since there's already a ticket to add IWYU support, I figure we can clean this up at the same time.

Reply:
Yeah SGTM; thanks for the comment on https://viam.atlassian.net/browse/RSDK-5479.